### PR TITLE
doc, refactor: changing -torcontrol help to specify that a default port is used

### PR DESCRIPTION
**Pull request by kevkevinpal**
Right now when we get the help for -torcontrol it says that there is a default ip and port we dont specify if there is a specified ip that we would also use port 9051 as default

Also I create a new const instead of using 9051 directly in the function

linking this PR because this was discussed here https://github.com/bitcoin/bitcoin/pull/28018

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -508,7 +508,7 @@ void SetupServerArgs(ArgsManager& argsman)
...
     argsman.AddArg("-timeout=<n>", strprintf("Specify socket connection timeout in milliseconds. If an initial attempt to connect is unsuccessful after this amount of time, drop it (minimum: 1, default: %d)", DEFAULT_CONNECT_TIMEOUT), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
     argsman.AddArg("-peertimeout=<n>", strprintf("Specify a p2p connection timeout delay in seconds. After connecting to a peer, wait this amount of time before considering disconnection based on inactivity (minimum: 1, default: %d)", DEFAULT_PEER_CONNECT_TIMEOUT), ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::CONNECTION);
-    argsman.AddArg("-torcontrol=<ip>:<port>", strprintf("Tor control port to use if onion listening enabled (default: %s)", DEFAULT_TOR_CONTROL), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
+    argsman.AddArg("-torcontrol=<ip>:<port>", strprintf("Tor control host and port to use if onion listening enabled (default: %s) If no port specified default port %s is used", DEFAULT_TOR_CONTROL, ToString(DEFAULT_TOR_CONTROL_PORT)), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
```
Comment (luke-jr) [inline src/init.cpp:511]: nit: I feel like this could be formatted better. Might need a comma after "specified".

**Review by luke-jr — APPROVED** [review]
utACK

File: src/init.cpp (hunk 1 above)
Comment (kevkevinpal) [reply src/init.cpp:511]: Does this look good?
I used chatgpt to refine the sentence a bit more to this

"Specify the Tor control host and optional port to use when onion listening is enabled (default: %s). If no port is specified, the default port will be used (default: %s)."

File: src/init.cpp [hunk 2]
```diff
@@ -508,7 +508,7 @@ void SetupServerArgs(ArgsManager& argsman)
...
     argsman.AddArg("-timeout=<n>", strprintf("Specify socket connection timeout in milliseconds. If an initial attempt to connect is unsuccessful after this amount of time, drop it (minimum: 1, default: %d)", DEFAULT_CONNECT_TIMEOUT), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
     argsman.AddArg("-peertimeout=<n>", strprintf("Specify a p2p connection timeout delay in seconds. After connecting to a peer, wait this amount of time before considering disconnection based on inactivity (minimum: 1, default: %d)", DEFAULT_PEER_CONNECT_TIMEOUT), ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::CONNECTION);
-    argsman.AddArg("-torcontrol=<ip>:<port>", strprintf("Tor control port to use if onion listening enabled (default: %s)", DEFAULT_TOR_CONTROL), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
+    argsman.AddArg("-torcontrol=<ip>:<port>", strprintf("Specify the Tor control host and optional port to use when onion listening is enabled (default: %s). If no port is specified, the default port will be used (default: %s).", DEFAULT_TOR_CONTROL, ToString(DEFAULT_TOR_CONTROL_PORT)), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
```
Comment (luke-jr) [inline src/init.cpp:511]: >I used chatgpt to refine the sentence a bit more to this

Please redo this without ChatGPT involved. LLMs do not have a clear copyright status.

Comment (kevkevinpal) [reply src/init.cpp:511]: Valid point will change to the following

`"Tor control host and port to use if onion listening enabled (default: %s). If no port is specified, the default port will be used (default: %s)."`

**Comment by kevkevinpal** [comment]
rebased to [5fa09d0](https://github.com/bitcoin/bitcoin/pull/28101/commits/5fa09d039f4e920c18f089fcb1fb4df7813468fc)

**Comment by kevkevinpal** [comment]
> Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

rebased properly in [5d94b1f](https://github.com/bitcoin/bitcoin/pull/28101/commits/5d94b1faecd8fc98017510d86f6ea5dc002ec0a3)

File: src/torcontrol.h [hunk 3]
```diff
@@ -19,6 +19,7 @@
 #include <string>
 #include <vector>
 
+static const int DEFAULT_TOR_CONTROL_PORT = 9051;
```
[the remark this answers is not in the document]
Comment (kevkevinpal) [reply src/torcontrol.h]: thanks didnt know that fixed in

[5d94b1f](https://github.com/bitcoin/bitcoin/pull/28101/commits/5d94b1faecd8fc98017510d86f6ea5dc002ec0a3)

File: src/init.cpp [hunk 4]
```diff
@@ -509,7 +509,7 @@ void SetupServerArgs(ArgsManager& argsman)
...
     argsman.AddArg("-timeout=<n>", strprintf("Specify socket connection timeout in milliseconds. If an initial attempt to connect is unsuccessful after this amount of time, drop it (minimum: 1, default: %d)", DEFAULT_CONNECT_TIMEOUT), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
     argsman.AddArg("-peertimeout=<n>", strprintf("Specify a p2p connection timeout delay in seconds. After connecting to a peer, wait this amount of time before considering disconnection based on inactivity (minimum: 1, default: %d)", DEFAULT_PEER_CONNECT_TIMEOUT), ArgsManager::ALLOW_ANY | ArgsManager::DEBUG_ONLY, OptionsCategory::CONNECTION);
-    argsman.AddArg("-torcontrol=<ip>:<port>", strprintf("Tor control port to use if onion listening enabled (default: %s)", DEFAULT_TOR_CONTROL), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
+    argsman.AddArg("-torcontrol=<ip>:<port>", strprintf("Tor control host and port to use if onion listening enabled (default: %s). If no port is specified, the default port will be used (default: %s).", DEFAULT_TOR_CONTROL, ToString(DEFAULT_TOR_CONTROL_PORT)), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
```
[the remark this answers is not in the document]
Comment (kevkevinpal) [reply src/init.cpp]: Update in e5f20cb26848c97d48c6a479c4680f6823f2ec1e

**Review by jonatack — COMMENTED** [review]
ACK e5f20cb26848c97d48c6a479c4680f6823f2ec1e

It might be good to prefix the pull title with `doc, refactor:` instead of `init`.

File: src/torcontrol.h [hunk 5]
```diff
@@ -19,6 +19,8 @@
 #include <string>
 #include <vector>
 
+
```
[the remark this answers is not in the document]
Comment (kevkevinpal) [reply src/torcontrol.h]: removed new line and updated title in [9a84200](https://github.com/bitcoin/bitcoin/pull/28101/commits/9a84200cfc994eebf38c46919b20e0c0261799ae)

**Review by MarnixCroes — COMMENTED** [review]
utACK 9a84200cfc994eebf38c46919b20e0c0261799ae

**Review by kristapsk — APPROVED** [review]
utACK 9a84200cfc994eebf38c46919b20e0c0261799ae

**Comment by achow101** [comment]
ACK 9a84200cfc994eebf38c46919b20e0c0261799ae

**Pull request merged**